datafusion/physical-expr/src/aggregate/mod.rs

```diff
@@ -77,6 +77,9 @@ pub trait AggregateExpr: Send + Sync + Debug {
     /// Single-column aggregations such as `sum` return a single value, others (e.g. `cov`) return many.
     fn expressions(&self) -> Vec<Arc<dyn PhysicalExpr>>;
 
+    /// FILTER (WHERE clause) expression for this aggregate
```

Comment:
I think it makes more sense to track the filters on the `LogicalPlan::GroupBy`, and relevant ExecutionPlan, rather than force all AggregateExprs to carry the filter themselves because"

1. The filtering is the same for all aggregates (it doesn't vary by aggregate type) so having the filter on the aggregate seems to be a mismatch
2. If forces user defined aggregates to all do the same thing (carry along a filter) and if they make a mistake they could get wrong answers.